service/storage_service.cc

```diff
@@ -2922,8 +2922,15 @@ future<> storage_service::join_token_ring(sharded<db::system_distributed_keyspac
             slogger.info("Replacing a node with {} IP address, my address={}, node being replaced={}",
                 get_broadcast_address() == *replace_address ? "the same" : "a different",
                 get_broadcast_address(), *replace_address);
-            tmptr->update_topology(*replace_address, std::move(ri->dc_rack), locator::node::state::being_replaced);
+            const auto replace_id = replacing_a_node_with_same_ip
+                ? tmptr->get_new()->get_my_id()
```

Comment:
I don't get this. The ID of replaced node should always be `ri->host_id` (i.e. the one we got from the cluster), no?

Why do we use *our own ID* as the replaced ID in some cases? Why does it matter if we replace with the same IP or not?